src/ads1x15/ads1015.cxx

```diff
@@ -55,6 +56,46 @@ ADS1015::ADS1015(int bus, uint8_t address, float vref) : ADS1X15(bus, address) {
         setGain(GAIN_SIXTEEN);
 }
 
+ADS1015::ADS1015(std::string initStr) : ADS1X15(initStr)
+{
+    float vref;
+    m_name = "ADS1015";
+    m_conversionDelay = ADS1015_CONVERSIONDELAY;
+    m_bitShift = 4;
+    ADS1X15::getCurrentConfig();
+
+    std::string leftoverString = ADS1X15::getLeftoverStr();
+    std::vector<std::string> upmTokens;
+
+    if(!leftoverString.empty())
+    {
+        upmTokens = UpmStringParser::parse(mraaIo->getLeftoverStr());
+    }
+    for (std::string tok : upmTokens)
```

Comment:
What happens with vref(being tested against multiple values below) if there is no leftover string?